lib/runtime/interface.go

```diff
@@ -12,6 +12,7 @@ import (
 )
 
 //go:generate mockery --name Instance --structname Instance --case underscore --keeptree
+//go:generate mockgen -destination=mocks/mock_instance.go -package=$GOPACKAGE . Instance
```

Comment:
I'd suggest to generate the mock as a `mock_instance_test.go` where it's needed instead of having a shared exported non-test one (to reduce go api noise, not affect code coverage and to reduce package dependency)

Reply:
I need help with this, when trying to change this I messed up mock references.

Reply:
In your case you use `mocks.NewMockInstance` in `dot/sync/chain_processor_test.go`, so:

1. remove that go generate line from `lib/runtime/interface.go`
2. remove the `mocks/mock_instance.go` file
3. Add to `dot/sync/chain_processor_test.go`

    ```go
    //go:generate mockgen -destination=mock_instance_test.go -package=$GOPACKAGE . Instance
    ```

4. Run `go generate ./dot/sync/...`
5. Change calls from `mocks.NewMockInstance` to `NewMockInstance` in `dot/sync/chain_processor_test.go`

Reply:
I also had issues doing this, after running go generate, I got this error: 
``` 
./prog.go:27:39: undefined: sync.Instance
prog.go:14:2: no required module provides package github.com/ChainSafe/gossamer/dot/sync: go.mod file not found in current directory or any parent directory; see 'go help modules'
prog.go:12:2: no required module provides package github.com/golang/mock/mockgen/model: go.mod file not found in current directory or any parent directory; see 'go help modules'
2022/04/25 17:36:53 Loading input failed: exit status 1
``` 
So I changed go generate to add source `//go:generate mockgen -destination=mock_instance_test.go -source=../../lib/runtime/interface.go -package=$GOPACKAGE . Instance` now go generate runs, but now when compiling I get errors 
```
./mock_interface_test.go:566:6: MockTransactionState redeclared in this block
    ./mock_instance_test.go:809:6: other declaration of MockTransactionState
./mock_interface_test.go:572:6: MockTransactionStateMockRecorder redeclared in this block
    ./mock_instance_test.go:815:6: other declaration of MockTransactionStateMockRecorder
./mock_interface_test.go:577:6: NewMockTransactionState redeclared in this block
    ./mock_instance_test.go:820:6: other declaration of NewMockTransactionState
./mock_interface_test.go:584:32: MockTransactionState.EXPECT redeclared in this block
    ./mock_instance_test.go:827:32: other declaration of EXPECT
```
I seems that `TransactionState` is defined in runtime/instance.go and sync/interface.go, thus the conflict.  I'm going to revert this for now, let me know if there is a better way to deal with this.

Reply:
@edwardmack try with `//go:generate mockgen -destination=mock_instance_test.go -package=$GOPACKAGE github.com/ChainSafe/gossamer/lib/runtime/interface Instance` instead of `-source` flag

Reply:
@EclesioMeloJunior that doesn't seem to work, when I update I get this error:
```
emack@system76-pc:~/projects/ChainSafe/gossamer$ go generate ./dot/sync/...
prog.go:14:2: no required module provides package github.com/ChainSafe/gossamer/lib/runtime/interface; to add it:
        go get github.com/ChainSafe/gossamer/lib/runtime/interface
prog.go:14:2: no required module provides package github.com/ChainSafe/gossamer/lib/runtime/interface: go.mod file not found in current directory or any parent directory; see 'go help modules'
prog.go:12:2: no required module provides package github.com/golang/mock/mockgen/model: go.mod file not found in current directory or any parent directory; see 'go help modules'
2022/05/02 16:52:08 Loading input failed: exit status 1
dot/sync/chain_processor_test.go:23: running "mockgen": exit status 1
```

Reply:
@EclesioMeloJunior yes, that fixed it!  Thanks.